clippy_lints/src/utils/author.rs

```diff
@@ -190,43 +195,343 @@ impl PrintVisitor {
+
+                println!("    if let LitKind::Str(ref {}, _) = {}.node;", str_pat, lit_pat);
+                println!("    if {}.as_str() == {:?}", str_pat, &*text.as_str());
+            },
+        }
+    }
+
+    fn print_loop(&mut self, body: &Block<'_>, label: Option<Label>, des: LoopSource) {
+        if_chain! {
+            if let LoopSource::While = des;
+            if let Some(body_expr) = body.expr;
+            if let ExprKind::If(if_expr, if_body, _) = body_expr.kind;
+            then {
+                match if_expr.kind {
+                    ExprKind::DropTemps(cond) => {
+                        let condition_pat = self.next("condition");
+                        let body_pat = self.next("body");
+
+                        println!("    if let Some(While {{ {} @ condition, {} @ body }}) = higher::While::hir({})", condition_pat, body_pat, self.current);
+
+                        self.current = condition_pat;
+                        self.visit_expr(cond);
+
+                        self.current = body_pat;
+                        self.visit_expr(body_expr);
+                    },
+                    ExprKind::Let(pat, expr, _) => {
+                        let let_pat = self.next("let_pat");
+                        let let_expr_pat = self.next("let_expr");
+                        let if_then_pat = self.next("if_then");
+
+                        println!("    if let Some(WhileLet {{ {} @ let_pat, {} @ let_expr, {} @ if_then }}) = higher::WhileLet::hir({})", let_pat, let_expr_pat, if_then_pat, self.current);
+
+                        self.current = let_pat;
+                        self.visit_pat(pat);
+
+                        self.current = let_expr_pat;
+                        self.visit_expr(expr);
+
+                        self.current = if_then_pat;
+                        self.visit_expr(if_body);
+                    },
+                    _ => {}
+                }
+            }
+        }
+
+        let body_pat = self.next("body");
+        let label_pat = self.next("label");
+
+        println!(
+            "    if let ExprKind::Loop(ref {}, ref {}, LoopSource::{:?}) = {};",
+            body_pat, label_pat, des, self.current
+        );
+
+        self.current = body_pat;
+        self.visit_block(body);
+
+        self.current = label_pat;
+        self.print_label(label);
+    }
+
+    fn print_if_expr(&mut self, cond: &Expr<'_>, then: &Expr<'_>, opt_else: Option<&Expr<'_>>) {
+        let cond_pat = self.next("cond");
+        let then_pat = self.next("then");
+        let else_pat = self.next("else_expr");
+
+        let else_ = if opt_else.is_some() {
+            format!("Some({})", else_pat)
+        } else {
+            String::from("None")
+        };
+
+        match cond.kind {
+            ExprKind::DropTemps(cond) => {
+                println!(
+                    "    if let Some(If {{ {} @ cond, {} @ then, r#else: {}}}) = higher::If::hir({})",
```

Comment:
I think this notation is more common for field bindings.

```suggestion
                    "    if let Some(If {{ cond: {}, {} @ then, r#else: {}}}) = higher::If::hir({})",
```